include/nbl/builtin/hlsl/bxdf/transmission/smooth_dielectric.hlsl

```diff
@@ -0,0 +1,332 @@
+    using isocache_type = IsoCache;
+    using anisocache_type = AnisoCache;
+
+    using params_isotropic_t = SmoothDielectricParams<LS, Iso, scalar_type>;
+    using params_anisotropic_t = SmoothDielectricParams<LS, Aniso, scalar_type>;
+
+
+    static this_t create(NBL_CONST_REF_ARG(spectral_type) eta2, NBL_CONST_REF_ARG(spectral_type) luminosityContributionHint)
+    {
+        this_t retval;
+        retval.eta2 = eta2;
+        retval.luminosityContributionHint = luminosityContributionHint;
+        return retval;
+    }
+
+    static this_t create(NBL_CONST_REF_ARG(SBxDFCreationParams<scalar_type, spectral_type>) params)
+    {
+        return create(params.eta2, params.luminosityContributionHint);
+    }
+
+    void init(NBL_CONST_REF_ARG(SBxDFCreationParams<scalar_type, spectral_type>) params)
+    {
+        eta2 = params.eta2;
+        luminosityContributionHint = params.luminosityContributionHint;
+    }
+
+    spectral_type eval(NBL_CONST_REF_ARG(params_isotropic_t) params)
+    {
+        return (spectral_type)0;
+    }
+    spectral_type eval(NBL_CONST_REF_ARG(params_anisotropic_t) params)
+    {
+        return (spectral_type)0;
+    }
+
+    // usually `luminosityContributionHint` would be the Rec.709 luma coefficients (the Y row of the RGB to CIE XYZ matrix)
+    // its basically a set of weights that determine
+    // assert(1.0==luminosityContributionHint.r+luminosityContributionHint.g+luminosityContributionHint.b);
+    // `remainderMetadata` is a variable which the generator function returns byproducts of sample generation that would otherwise have to be redundantly calculated `quotient_and_pdf`
+    sample_type __generate_wo_clamps(NBL_CONST_REF_ARG(vector3_type) V, NBL_CONST_REF_ARG(vector3_type) T, NBL_CONST_REF_ARG(vector3_type) B, NBL_CONST_REF_ARG(vector3_type) N, scalar_type NdotV, scalar_type absNdotV, NBL_REF_ARG(vector3_type) u, NBL_CONST_REF_ARG(spectral_type) eta2, NBL_CONST_REF_ARG(spectral_type) luminosityContributionHint, NBL_REF_ARG(spectral_type) remainderMetadata)
+    {
+        // we will only ever intersect from the outside
+        const spectral_type reflectance = fresnel::thinDielectricInfiniteScatter<spectral_type>(fresnel::Dielectric<spectral_type>::__call(eta2,absNdotV));
+
+        // we are only allowed one choice for the entire ray, so make the probability a weighted sum
+        const scalar_type reflectionProb = nbl::hlsl::dot<spectral_type>(reflectance, luminosityContributionHint);
+
+        scalar_type rcpChoiceProb;
+        const bool transmitted = math::partitionRandVariable(reflectionProb, u.z, rcpChoiceProb);
+        remainderMetadata = (transmitted ? ((spectral_type)(1.0) - reflectance) : reflectance) * rcpChoiceProb;
+
+        ray_dir_info_type L;
+        L.direction = (transmitted ? (vector3_type)(0.0) : N * 2.0f * NdotV) - V;
+        return sample_type::create(L, nbl::hlsl::dot<vector3_type>(V, L.direction), T, B, N);
```

Comment:
`VdotL` can be computed much more cheaply as `(transmitted ? 0.f:2.f)*NdotV*NdotV-1.f;`